frontend/src/utils/ApiService.js

```
@@ -1,224 +1,458 @@
+            }
+          };
+        }
+
+        xhr.send(formData);
+      } catch (error) {
+        console.error('Error setting up file upload:', error);
+        reject(error);
       }
-      
-      xhr.send(formData);
     });
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)